### PR TITLE
Bugfix - Retrieve Settings from KV map

### PR DESCRIPTION
**Pull request by beejones**
There was a bug in retrieving the settings from the KV map which is fixed. As a result the default was always used which resulted in debug log not being logged.
New unit tests and pytest added
Added unit tests command to makefile

### CODE REVIEW

**Comment by beejones** [comment]
tests:
https://github.com/microsoft/privacy-sandbox-dev/actions/runs/11935821287

**Comment by DomAyre** [comment]
It looks like we have added a new endpoint, regarding that: 
- Based on the test expectation, the output looks like it returns very similar to `/heartbeat` what is the distinction?
- Can we add system tests for it? I tried to use it locally, not sure if I'm missing something? I ran `/heartbeat` after to ensure the network was up correctly 
```
@DomAyre ➜ /workspaces/azure-privacy-sandbox-kms (test-ronny) $ make start-host-idp
stop-host
source ./scripts/ccf/sandbox-local/down.sh
stop-idp
source ./scripts/jwt-issuer/down.sh
[+] Running 1/0
 ✔ Container services-jwt-issuer-1  Removed                                                                                                                       0.0s 
start-idp
source ./scripts/jwt-issuer/up.sh
[+] Running 1/1
 ✔ Container services-jwt-issuer-1  Healthy                                                                                                                       2.0s 
start-host
MEMBER_COUNT=3 source ./scripts/ccf/sandbox-local/up.sh && \
source ./scripts/kms/js-app-set.sh && \
source ./scripts/kms/constitution-set.sh ./governance/constitution/kms_actions.js
[+] Running 1/1
 ✔ Container services-ccf-sandbox-1  Healthy                                                                                                                      3.1s 
{
  "WORKSPACE": "/workspaces/azure-privacy-sandbox-kms/workspace",
  "KMS_URL": "https://127.0.0.1:8000",
  "KMS_SERVICE_CERT_PATH": "/workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/service_cert.pem",
  "KMS_MEMBER_CERT_PATH": "/workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_cert.pem",
  "KMS_MEMBER_PRIVK_PATH": "/workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_privk.pem"
}

up to date, audited 832 packages in 4s

88 packages are looking for funding
  run `npm fund` for details

4 vulnerabilities (3 low, 1 high)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

> build
> rm -rf ./dist/ && rollup --config && cp app.json dist/ && node build_bundle.js dist/


src/index.ts → dist/src...
created dist/src in 6.3s
Writing bundle containing 63 modules to dist/bundle.json
Proposing: /workspaces/azure-privacy-sandbox-kms/workspace/proposals/set_js_app.json
  to https://127.0.0.1:8000
    cert: /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/service_cert.pem
  as /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_cert.pem
{
  "ballot_count": 0,
  "proposal_id": "d86d3f3ddd6d7d18f1b514767dd006577cd4a7a7c035779072ab5b18809fc168",
  "proposer_id": "cff9ff64886712c161c63dd118a9be6ea19f4bc289f6a72fc9c325caaaae0fea",
  "state": "Accepted",
  "votes": {}
}
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 48068  100 48068    0     0  1738k      0 --:--:-- --:--:-- --:--:-- 1805k
Proposing: /workspaces/azure-privacy-sandbox-kms/workspace/proposals/set_constitution.json
  to https://127.0.0.1:8000
    cert: /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/service_cert.pem
  as /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_cert.pem
{
  "ballot_count": 0,
  "proposal_id": "10558746788b5e5b9deeefed6db3b51ada56e5ae3d1cf33f76903c29734db7c4",
  "proposer_id": "cff9ff64886712c161c63dd118a9be6ea19f4bc289f6a72fc9c325caaaae0fea",
  "state": "Accepted",
  "votes": {}
}
start-host-idp
Executing: 
MEMBER_COUNT=3 source ./scripts/ccf/sandbox-local/up.sh > /dev/null 2>&1 && \
source ./scripts/kms/jwt-issuer-trust.sh
Proposing: /workspaces/azure-privacy-sandbox-kms/workspace/proposals/set_jwt_issuer.json
  to https://127.0.0.1:8000
    cert: /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/service_cert.pem
  as /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_cert.pem
{
  "ballot_count": 0,
  "proposal_id": "0d6fef71560b991c2871d89bd6c669309aac42a34a90124595543107ad13c56b",
  "proposer_id": "cff9ff64886712c161c63dd118a9be6ea19f4bc289f6a72fc9c325caaaae0fea",
  "state": "Accepted",
  "votes": {}
}
@DomAyre ➜ /workspaces/azure-privacy-sandbox-kms (test-ronny) $ curl -k https://127.0.0.1:8000/app/settingsPolicy --cacert /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/service_cert.pem --cert /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_cert.pem --key /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_privk.pem -H "Content-Type: application/json"
{"error":{"code":"InternalError","message":"Exception thrown while executing."}}
@DomAyre ➜ /workspaces/azure-privacy-sandbox-kms (test-ronny) $ curl -k https://127.0.0.1:8000/app/heartbeat --cacert /workspaces/azure-privacy-sandbox-kms/workspace/s
andbox_common/service_cert.pem --cert /workspaces/azure-privacy-sandbox-kms/workspace/sandbox_common/member0_cert.pem --key /workspaces/azure-privacy-sandbox-kms/works
pace/sandbox_common/member0_privk.pem -H "Content-Type: application/json"
{"description":["azure-privacy-sandbox-kms","Key Management Service","1.0.0","false"]}
```

File: Makefile [hunk 1]
```diff
@@ -198,6 +198,8 @@ release-policy-set:
 	KMS_URL=${KMS_URL} \
 	RELEASE_POLICY_PROPOSAL=$(release-policy-proposal) \
 		./scripts/release-policy-set.sh
+test-unit:
```
Comment (DomAyre) [inline Makefile:201]: nit: spacing

**Review by DomAyre — COMMENTED** [review]
Had a discussion about /heartbeat vs /settingsPolicy, /heartbeat can now return something simple like "Running" in json

File: scripts/settings-policy-set.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+REPO_ROOT="$(realpath "$(dirname "$(realpath "${BASH_SOURCE[0]}")")/..")"
+
+# Construct the proposal
+cp $REPO_ROOT/$SETTINGS_POLICY_PROPOSAL $WORKSPACE/proposals/set_settings_policy.json
```
Comment (DomAyre) [inline scripts/settings-policy-set.sh:8]: Would it be more flexible to allow settings policy path to not have to be relative to REPO_ROOT?

How about this? Then relative paths would still work for convenience 
```
(cd $REPO_ROOT && cp $SETTINGS_POLICY_PROPOSAL $WORKSPACE/proposals/set_settings_policy.json)
```

File: test/system-test/test_settingsPolicy.py
Comment (DomAyre) [inline test/system-test/test_settingsPolicy.py:1]: Good set of tests, can we add a test which sets the settings policy, then changes it to ensure we correctly overwrite the policy?

Comment (beejones) [reply test/system-test/test_settingsPolicy.py:1]: sure

File: test/system-test/test_settingsPolicy.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    assert status_code == 401
+
+
+def test_settingsPolicy_with_policy(setup_kms):
+    apply_kms_constitution(setup_kms["url"], setup_kms["workspace"])
+    apply_settings_policy(setup_kms["url"], setup_kms["workspace"])
+    status_code, settings_json = settingsPolicy(setup_kms["url"])
+    assert status_code == 200
+    assert settings_json == {
+    "service": {
```
[the remark this answers is not in the document]
Comment (beejones) [reply test/system-test/test_settingsPolicy.py:31]: debug is changed from false to true

File: test/system-test/test_settingsPolicy.py
Comment (DomAyre) [reply test/system-test/test_settingsPolicy.py:1]: The new test you added looks good but it isn't quite what I asked for, it still only sets the settings policy once, just like the test above. I was hoping that you would set the settings policy once and then change it

Comment (beejones) [reply test/system-test/test_settingsPolicy.py:1]: I added a change of settings

File: test/system-test/test_settingsPolicy.py [hunk 4]
```diff
@@ -66,7 +66,22 @@ def test_settingsPolicy_with_custom_policy(setup_kms):
...
+                "args": {
+                    "settings_policy":  settings_policy
+                }
+            }
+        ]
+    }
+    apply_settings_policy(setup_kms["url"], setup_kms["workspace"], proposal)
+    status_code, settings_json = settingsPolicy(setup_kms["url"])
+    assert status_code == 200
+    assert settings_json.get("service").get("debug") == False
```
Comment (DomAyre) [inline test/system-test/test_settingsPolicy.py:84]: why `get()` instead of direct key access?

File: test/system-test/test_settingsPolicy.py
Comment (DomAyre) [reply test/system-test/test_settingsPolicy.py:1]: Great thanks

**Review by DomAyre — APPROVED** [review]

**Pull request merged**